core/src/test/java/org/hibernate/ogm/sessionfactory/SessionFactoryInlineTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.sessionfactory;
+
+import static junit.framework.TestCase.fail;
+
+import java.util.function.Consumer;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+/**
+ * Helps to test {@link SessionFactory} creation
+ * providing an inline test API
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class SessionFactoryInlineTest {
+
+	/**
+	 * Expect {@link Exception} during the build of {@link SessionFactory}
+	 *
+	 * @param builder given test configuration
+	 * @param consumer function to apply to the expected exception
+	 */
+	public static void assertInitializationFailure(SessionFactoryBuilder builder, Consumer<Exception> consumer) {
+		try ( SessionFactory sessionFactory = builder.build() ) {
+			fail( "Expected exception at Hibernate factory creation time was not raised!" );
+		}
+		catch (Exception rootException) {
+			consumer.accept( rootException );
+		}
+	}
```

Comment:
This method doesn't seem all that useful either, why not call builder.build() in the test and then test the exception using the rule form JUnit?